### PR TITLE
cleanup: Remove deprecated modules

### PR DESCRIPTION
**Pull request by Reto911**
In this PR, we removed some modules that will never be used:
- Distributed cache, in `async_fuse/memfs/dist`, and the file `id_alloc.rs` is moved out of that dir. (#433)
- Old `S3Backend`, which is replaced by new `Backend` implemented with `openDAL`. By the way, the dependency `rust-s3` is also removed.
- Old `GlobalCache`, which is replaced by new `MemoryCache`.

Besides, we moved `async_fuse/memfs/cache` to the root of sources, and renamed it to `storage`.

### CODE REVIEW

File: src/async_fuse/memfs/s3_metadata.rs [hunk 1]
```diff
@@ -54,26 +48,19 @@ macro_rules! build_inconsistent_fs {
 const MY_TTL_SEC: u64 = 3600; // TODO: should be a long value, say 1 hour
 /// The generation ID of FUSE attributes
 const MY_GENERATION: u64 = 1; // TODO: find a proper way to set generation
-#[allow(dead_code)]
 /// The limit of transaction commit retrying times.
 const TXN_RETRY_LIMIT: u32 = 10;
 
 /// File system in-memory meta-data
 #[derive(Debug)]
 #[allow(dead_code)]
```
Comment (xiaguan) [inline src/async_fuse/memfs/s3_metadata.rs:56]: Is this mark still available?

File: src/async_fuse/memfs/s3_node.rs [hunk 2]
```diff
@@ -45,7 +40,7 @@ pub enum S3NodeData {
     /// Directory entry data
     Directory(BTreeMap<String, DirEntry>),
     /// File content data
-    RegFile(Arc<GlobalCache>),
+    RegFile,
```
Comment (xiaguan) [inline src/async_fuse/memfs/s3_node.rs:43]: After dir entry refactor, the `NodeData` seems meaning less :thinking:

File: src/async_fuse/memfs/s3_node.rs [hunk 3]
```diff
@@ -62,17 +57,15 @@ impl S3NodeData {
...
         }
     }
 }
 
 /// A file node or a directory node
 #[derive(Debug)]
-pub struct S3Node<S: S3BackEnd + Sync + Send + 'static> {
-    /// S3 Backend
-    s3_backend: Arc<S>,
+pub struct S3Node {
```
Comment (xiaguan) [inline src/async_fuse/memfs/s3_node.rs:68]: We should rename `S3Metadata` and `S3Node` since they have nothing to do with "S3".

**Comment by xiaguan** [comment]
This pull request is a simple code structure refactor.  LGTM.

File: src/async_fuse/memfs/s3_node.rs (hunk 2 above)
Comment (Reto911) [reply src/async_fuse/memfs/s3_node.rs:43]: Yes, the only thing it holds is the target path of symlink.

File: src/async_fuse/memfs/s3_node.rs (hunk 3 above)
Comment (Reto911) [reply src/async_fuse/memfs/s3_node.rs:68]: Make sense. And the trait of `Metadata` and `Node` should be considered further.

File: src/async_fuse/memfs/s3_metadata.rs (hunk 1 above)
Comment (Reto911) [reply src/async_fuse/memfs/s3_metadata.rs:56]: Yes, the field named `cur_fd` is never used. I don't know what is it.

**Review by xiaguan — APPROVED** [review]
LGTM

**Pull request merged**